src/libsystemd-network/network-common.c

```diff
@@ -13,3 +14,90 @@ int get_ifname(int ifindex, char **ifname) {
 
         return format_ifname_alloc(ifindex, ifname);
 }
+
+usec_t unaligned_be32_sec_to_usec(const void *p, bool max_as_infinity) {
+        uint32_t s = unaligned_read_be32(ASSERT_PTR(p));
+
+        if (s == UINT32_MAX && max_as_infinity)
+                return USEC_INFINITY;
+
+        return s * USEC_PER_SEC;
+}
+
+usec_t be32_sec_to_usec(be32_t t, bool max_as_infinity) {
+        uint32_t s = be32toh(t);
+
+        if (s == UINT32_MAX && max_as_infinity)
+                return USEC_INFINITY;
+
+        return s * USEC_PER_SEC;
+}
+
+usec_t be32_msec_to_usec(be32_t t, bool max_as_infinity) {
+        uint32_t s = be32toh(t);
+
+        if (s == UINT32_MAX && max_as_infinity)
+                return USEC_INFINITY;
+
+        return s * USEC_PER_MSEC;
+}
+
+usec_t be16_sec_to_usec(be16_t t, bool max_as_infinity) {
+        uint16_t s = be16toh(t);
+
+        if (s == UINT16_MAX && max_as_infinity)
+                return USEC_INFINITY;
+
+        return s * USEC_PER_SEC;
+}
+
+be32_t usec_to_be32_sec(usec_t t) {
+        if (t == USEC_INFINITY)
+                /* UINT32_MAX may be handled as infinity. */
+                return htobe32(UINT32_MAX);
+
+        assert(t <= UINT32_MAX * USEC_PER_SEC);
```

Comment:
if you generalize these in helpers it might make sense to saturate the values if out of range rather than hit an assert.

(i.e. if the helpers are generalized and in a different .c file it becomes harder to reivew that all input requirements are kept when calling them, hence I think it would be wise to just saturate here so that something reasonable happens.)

Could be as easy as changing the `t == USEC_INFINITY` check above to `t >= UINT32_MAX*USEC_PERSEC)`